bin/pasta_file_change_recipients.py

```diff
@@ -0,0 +1,108 @@
+"""
+PaStA - Patch Stack Analysis
+
+Author:
+  Shubhamkumar Pandey <b18194@students.iitmandi.ac.in>
+
+This work is licensed under the terms of the GNU GPL, version 2.  See
+the COPYING file in the top-level directory.
+"""
+
+import os
+import sys
+import argparse
+
+from tqdm import tqdm
+from collections import defaultdict, Counter
+from logging import getLogger
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+from pypasta import *
+from pypasta.LinuxMailCharacteristics import email_get_recipients
+from pypasta.Repository.Patch import Diff
+
+log = getLogger(__name__[-15:])
+
+RECIPIENTS = None
+
+def dump_csv(config, f_recipients):
+    repo = config.repo
+    global RECIPIENTS
+    RECIPIENTS = defaultdict(Counter)
+    for message_id in tqdm(repo.mbox.get_ids(time_window=config.mbox_time_window)):
+        patch = repo[message_id]
+        mail = repo.mbox.get_messages(message_id)[0]
+        recipients = email_get_recipients(mail)
+        for filename in patch.diff.affected:
+            for recipient in recipients:
+                RECIPIENTS[filename][recipient] += 1
+    
+    if f_recipients:
+        with open(f_recipients, 'w') as f:
+            f.write('filename, recipients, count\n')
+            for filename, ctr in RECIPIENTS.items():
+                ctr = sorted(ctr.items(), key = lambda x: x[1], reverse=True)
+                f.write(f"{filename}, {','.join(map(lambda x: x[0]+','+str(x[1]),ctr))}\n")
+                RECIPIENTS[filename] = set(map(lambda x: x[0], ctr))
```

Comment:
What happens in this line? I don't get it.

Reply:
Just writing to file in `filename,mail_id,count,mail_id,count...` format
and taking the RECIPIENT[filename] = set of all mail_ids.